lib/middleware/operation.js

```diff
@@ -32,55 +34,103 @@ function findPropertyOperations ({ types, method, term, resource }) {
   return operations
 }
 
-function factory (api) {
-  return async (req, res, next) => {
-    if (!req.hydra.resource) {
-      return next()
-    }
+function findPropertyOperations ({ term, resourceCandidates, api, method }) {
+  const apiGraph = clownface(api)
 
-    const method = req.method === 'HEAD' ? 'GET' : req.method
-    const types = clownface({ ...api, term: [...req.hydra.resource.types] })
+  return resourceCandidates
+    .reduce((matched, resource) => {
+      const types = apiGraph.node([...resource.types])
 
-    let operations
-    if (req.hydra.term.equals(req.hydra.resource.term)) {
-      // only look for direct operation when the root resource is requested
-      operations = findClassOperations(types, method)
-    }
-    if (!operations) {
-      // otherwise try finding the operation by property usage
-      operations = findPropertyOperations({ types, method, term: req.hydra.term, resource: req.hydra.resource })
-    }
+      const more = findCandidatePropertyOperations({ types, method, term, resource })
+
+      if (!matched) {
+        return more
+      }
+
+      if (more.terms.length === 0) {
+        return matched
+      }
 
-    const [operation, ...rest] = (operations || []).toArray()
+      return clownface({
+        _context: [...matched._context, more._context],
+      })
```

Comment:
Is this the correct way to merge multiple clownface contexts?

I use this way to combine all potential matching operations for the property candidates